google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -1269,4 +1271,26 @@ public void testInstanceReturnedByGetServiceCanBeUsedToDeserializeAQuery() throw
     fs.close();
     Query.fromProto(fs, proto);
   }
+
+  @Test
+  public void deleteNestedFieldUsingFieldPath() throws Exception {
+    DocumentReference documentReference = randomColl.document("doc1");
+    Map<String, Object> dotKeyMap = new HashMap<>();
+    dotKeyMap.put("a.b", SINGLE_FILED_MAP_WITH_DOT);
+    documentReference.set(dotKeyMap).get();
+    DocumentSnapshot documentSnapshots = documentReference.get().get();
+    assertEquals(SINGLE_FILED_MAP_WITH_DOT, documentSnapshots.getData().get("a.b"));
+    FieldPath path = FieldPath.of("a.b", "c.d");
+    documentReference.update(path, FieldValue.delete()).get();
+    documentSnapshots = documentReference.get().get();
+    assertNull(documentSnapshots.getData().get("c.d"));
+    Map<String, Object> nestedMap = new HashMap<>();
+    nestedMap.put("e", SINGLE_FILED_MAP_WITH_DOT);
+    dotKeyMap.put("a.b", nestedMap);
+    documentReference.set(dotKeyMap).get();
+    path = FieldPath.of("a.b", "e", "c.d");
+    documentReference.update(path, FieldValue.delete()).get();
+    documentSnapshots = documentReference.get().get();
+    assertNull(documentSnapshots.getData().get("c.d"));
```

Comment:
I think you can delete this part as it doesn't expand on the tests coverage.

Reply:
Done